### ISSUE TITLE
Show progress during parallel tuning

### ISSUE THREAD

**Issue by kevin-m-kent** [tidymodels/tune#335]
## Feature

In situations when you are running tune_grid in parallel, I'd like to be able to see the progress (either overall or worker level) of the tuning (at the very least a progress bar but would be nice to have the option of verbose output). For instance:

```
doParallel::registerDoParallel()

xgb_tune <-
  tune_grid(xgb_workflow, resamples = sample_folds, grid = xbg_grid, control = control_grid(verbose = TRUE, allow_par = TRUE,parallel_over = "everything"))
```
Would like to see verbose output like:

```
i Bootstrap01: preprocessor 1/1
v Bootstrap01: preprocessor 1/1
i Bootstrap01: preprocessor 1/1, model 1/5
v Bootstrap01: preprocessor 1/1, model 1/5
i Bootstrap01: preprocessor 1/1, model 1/5 (predictions)
i Bootstrap01: preprocessor 1/1, model 2/5
v Bootstrap01: preprocessor 1/1, model 2/5
```
or a progress bar like the progress package

```
[==========================================================-------------]  81%
```

or like spark
```
[Stage X:==========>            (A + B) / C]
```

**Comment by topepo** [comment]
The underlying problem is in how the parallel backends handle output. In many cases, they do not print the logging (but it is generated). Different backends do different things and the there are also differences in output between running in the terminal and running in the RStudio IDE. It's a bit of a mess. 

We will probably make an option in `tune` to use `future` (we currently use `foreach`) and that would enable better logging. I'm generally opposed to progress bars but we should be able to print out the current logging information.

**Comment by franzbischoff** [comment]
The package [progressr](https://github.com/HenrikBengtsson/progressr) is suggested by `furrr` as an API for progress outputs.
As @topepo pointed out, different backends have different ways to handle outputs. The `future` documentation has a nice explanation of how they make outputs "bubble up" to the current R session.

`progressr` uses handlers (callbacks) to receive updates. They also make clear that the type of progress output shall be set by the user and not by the packages (in this case, `tune`).

Maybe a custom "extract" function (parameter of `control_grid()` could be used to send updates to `progressr` for example?

My 2 cents 😃

**Comment by topepo** [comment]
Going to close this. If/when we get to using futures with tune, I'll reference this so that we can add progress.
